functorch/functorch/_src/aot_autograd.py

```diff
@@ -174,146 +175,192 @@ def call_func_with_args(f, args, steal_args=False):
     if not steal_args:
         args = list(args)
     assert isinstance(args, list)
-    # TODO: Please remove soon
-    # https://github.com/pytorch/pytorch/pull/83137#issuecomment-1211320670
+
+    # Disable the JIT Autocast flag to prevent re-autocasting of jitted graph.
+    # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
+    old_jit_autocast_flag = torch._C._jit_set_autocast_mode(False)
     if hasattr(f, "_boxed_call"):
-        return normalize_as_list(f(args))
+        out = normalize_as_list(f(args))
     else:
+        # TODO: Please remove soon
+        # https://github.com/pytorch/pytorch/pull/83137#issuecomment-1211320670
         logging.warn(
             "Your compiler for AOTAutograd is returning a a function that doesn't take boxed arguments. "
             "Please wrap it with functorch.compile.make_boxed_func or handle the boxed arguments yourself. "
             "See https://github.com/pytorch/pytorch/pull/83137#issuecomment-1211320670 for rationale."
         )
-        return normalize_as_list(f(*args))
+        out = normalize_as_list(f(*args))
+    torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
+    return out
 
 
-def create_aot_autograd_function(
-    flat_fn, fw_compiler, bw_compiler, partition_fn, decompositions, grad_state
-):
+@dataclasses.dataclass
+class AOTConfig:
+    """
+    Configuration for AOTDispatcher
     """
-    Traces the forward and backward graphs of the attr:`flat_fn` to generate a
-    joint graph. The joint graph is an Fx graph with Aten ops. Please refer to
-    the tracing mechanism to understand the graph capturing details.
 
-    The joint graph is then passed through attr:`partition_fn` to isolate the
-    forward and backward portions, which are then respectively compiled via the
-    provided attr:`fw_compiler` and attr:`bw_compiler`.
+    fw_compiler: Callable
+    bw_compiler: Callable
+    partition_fn: Callable
+    decompositions: Dict[Callable, Callable]
 
-    The resulting compiled forward and backward graphs are then wrapped up in a
-    ``torch.autograd.Function`` object.
-    """
-    if decompositions is None:
-        decompositions = {}
+
+def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):
+    def fake_signature(fn, nargs):
+        """FX gets confused by varargs, de-confuse it"""
+        argnames = ",".join(f"arg{i}" for i in range(nargs))
+        return eval(f"lambda {argnames}: fn({argnames})", {"fn": fn})
```

Comment:
for my understanding: the way this is de-confusing it is... passing in the args one at a time? The eval feels hacky, how difficult would it be to teach FX about varargs?